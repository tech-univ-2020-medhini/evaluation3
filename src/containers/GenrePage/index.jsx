import React from 'react';
import PropTypes from 'prop-types';
import GenreCard from '../../components/GenreCard';

const GenrePage = (props) => {
  const { songsList, setSongsOfGenre } = props;
  const getSongs = (genre) => {
    console.log(genre);
    const genreSongs = songsList.filter((song) => {
      const { genres } = song;
      if (genres.indexOf(genre) !== -1) {
        return true;
      }
      return false;
    });
    setSongsOfGenre(genreSongs);
  };
  //   const genres = genreList.map((genre) => (
  //     <GenreCard name={genre} imageLink="" getSongs={getSongs} />
  //   ));

  return (
    <div className="Genre-page">
      <p>
        Genres
      </p>
      <div className="Genre-container">
        <GenreCard name="rock" getSongs={() => getSongs('rock')} />
        <GenreCard name="pop" getSongs={() => getSongs('pop')} />
        <GenreCard name="bollywood" getSongs={() => getSongs('bollywood')} />
        <GenreCard name="country" getSongs={() => getSongs('country')} />
      </div>
    </div>
  );
};
GenrePage.propTypes = {
  songsList: PropTypes.arrayOf(PropTypes.object).isRequired,
  setSongsOfGenre: PropTypes.func.isRequired,
};
export default GenrePage;
