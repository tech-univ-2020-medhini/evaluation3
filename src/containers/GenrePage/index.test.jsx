import React from 'react';
import { render } from '@testing-library/react';
import GenrePage from './index';


describe('the genre page container', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<GenrePage />);

    expect(asFragment()).toMatchSnapshot();
  });
});
