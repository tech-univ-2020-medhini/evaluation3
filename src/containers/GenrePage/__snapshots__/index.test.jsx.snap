// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the genre page container should render correctly 1`] = `
<DocumentFragment>
  <div
    class="Genre-page"
  >
    <p>
      Genres
    </p>
    <div
      class="Genre-container"
    >
      <div
        class="GenreCard"
      >
        <img
          alt="/"
        />
        <div
          class="Button"
        >
          <button
            data-testid="test-btn"
            type="button"
          >
            rock
          </button>
        </div>
      </div>
      <div
        class="GenreCard"
      >
        <img
          alt="/"
        />
        <div
          class="Button"
        >
          <button
            data-testid="test-btn"
            type="button"
          >
            pop
          </button>
        </div>
      </div>
      <div
        class="GenreCard"
      >
        <img
          alt="/"
        />
        <div
          class="Button"
        >
          <button
            data-testid="test-btn"
            type="button"
          >
            bollywood
          </button>
        </div>
      </div>
      <div
        class="GenreCard"
      >
        <img
          alt="/"
        />
        <div
          class="Button"
        >
          <button
            data-testid="test-btn"
            type="button"
          >
            country
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
